tests/core/editor/detaching.js

```diff
@@ -0,0 +1,176 @@
+// jscs:enable maximumLineLength
+/* global detachingTools */
+
+( function() {
+	var detachWhenScriptLoaded = detachingTools.detachWhenScriptLoaded,
+		detachBeforeIframeLoad = detachingTools.detachBeforeIframeLoad,
+		fakeComponent = CKEDITOR.document.findOne( '#fake-component' ),
+		container = CKEDITOR.document.findOne( '#container' ),
+		editor, currentError;
+
+	window.onerror = function( error ) {
+		currentError = error;
+	};
+
+	fakeComponent.remove();
+	fakeComponent = fakeComponent.getOuterHtml();
+
+	// (#3115)
+	var tests = createDetatchTests(
+		[ {
+			name: 'test detach and destroy synchronously',
+			callback: function( detach ) {
+				detach();
+			}
+		}, {
+			name: 'test detach and destroy asynchronously',
+			callback: function( detach ) {
+				setTimeout( detach );
+			}
+		}, {
+			name: 'test detach before firing editor#loaded event',
+			callback: function( detach, editor ) {
+				editor.once( 'loaded', detach, null, null, -9999 );
+			}
+		}, {
+			name: 'test detach before scriptLoader.load fires it\'s callback',
+			callback: detachWhenScriptLoaded
+		}, {
+			name: 'test detach before iframe#onload',
+			callback: function( detach, editor ) {
+				detachBeforeIframeLoad( detach, editor );
+			},
+			ignore: !CKEDITOR.env.gecko && !CKEDITOR.env.ie || CKEDITOR.env.edge,
+			editor: 'classic'
+		} ]
+	);
+
+	// (#3115)
+	tests[ 'test editor set mode when editor is detached' ] = testSetMode( function( editor ) {
+		sinon.stub( editor.container, 'isDetached' ).returns( true );
+	} );
+
+	// (#3115)
+	tests[ 'test editor set mode when editor is destroyed' ] = testSetMode( function( editor ) {
+		editor.status = 'destroyed';
+	} );
+
+	bender.test( tests );
+
+	function createDetatchTests( tests ) {
+		var editors = [ 'classic', 'inline', 'divarea' ],
+			newTests = {};
+
+		forEach( tests, function( testOptions ) {
+			forEach( editors, function( editorName ) {
+				var tcName = testOptions.name + ' (' + editorName + ', ';
+
+				if ( testOptions.ignore ) {
+					return;
+				}
+
+				if ( testOptions.editor && testOptions.editor !== editorName ) {
+					return;
+				}
+
+				newTests[ tcName + 'div)' ] = testDetach( testOptions.callback, editorName, 'div' );
+				newTests[ tcName + 'textarea)' ] = testDetach( testOptions.callback, editorName, 'textarea' );
+			} );
+		} );
+
+		return newTests;
+	}
+
+	function testDetach( callback, editorName, elementName ) {
+		return function() {
+			var html = elementName === 'div' ? fakeComponent.replace( /textarea/g, 'div' ) : fakeComponent,
+				component = CKEDITOR.dom.element.createFromHtml( html ),
+				config = {};
+
+			container.append( component );
+
+			switch ( editorName ) {
```

Comment:
Is this `switch` even needed? It probably can be done in a simpler, more readable way, e.g.

```javascript
var config = editorName === 'divarea' ? { extraPlugins: 'divarea' } : {},
	editor = CKEDITOR[ editorName === 'inline' ? 'inline' : 'replace' ]( 'editor', config );
```

Additionally isn't overriding config with `config` variable also overriding all plugins loaded via `bender-ckeditor-plugins` comment?